mlir/lib/Conversion/RaiseWasm/RaiseWasmMLIR.cpp

```diff
@@ -42,19 +44,17 @@ using namespace mlir::wasm;
 
 namespace {
 
-template <typename SourceOp, typename TargetIntOp, typename TargetFPOp>
+template <typename SourceOp, typename TargetIntOp, typename TargetFPOp, typename IntrinsicOp>
 struct IntFPDispatchMappingConversion : OpConversionPattern<SourceOp> {
   using OpConversionPattern<SourceOp>::OpConversionPattern;
 
   LogicalResult
   matchAndRewrite(SourceOp srcOp, typename SourceOp::Adaptor adaptor,
                   ConversionPatternRewriter &rewriter) const override {
 
-    // If the op is trappable, do not lower it here. Leave it to the WasmMLIRToEmbedder
-    // pass to insert an intrinsic or an embedder specific call instead.
     auto interfaceOp = llvm::dyn_cast_or_null<WasmTrappableCheckOpInterface>(srcOp.getOperation());
     if (interfaceOp && llvm::succeeded(interfaceOp.shouldCreateTrap()))
-      return failure();
+      return handleTrappableOperation(srcOp, rewriter);
```

Comment:
following previous comment: that way we can get rid of all this internal pattern logic.

Rationale: for one operation we know if we need or not the intrinsic handling.